designs/SearchCommandSpec.md

```diff
@@ -0,0 +1,96 @@
+* Issue: [9704](https://github.com/NuGet/Home/issues/9704) - Search CLI Tool for NuGet.exe and Dotnet.exe
+
+## Problem Background
+
+Search functionality from the command line for NuGet.exe and the .NET Core CLI has been a feature request from the NuGet open source community for some time. Additionally, the NuGet API is already equipped to handle search requests, so an extension of this service to command line users would be both practical to implement and also highly valuable.
+
+## Who are the customers
+
+This targets all NuGet.exe and .NET users that want to be able to search for packages from the command line, rather than using NuGet Gallery or VS PMUI. Also, while `nuget.exe list` allows users to search different package sources for packages with a search query, the results are returned sorted by the package Id, rather than by relevance, as proposed for this Search command. Further, the Search command would provide more information on the packages than List currently does.
+
+## Goals
+
+* Providing search functionality for NuGet packages through the commandline for NuGet.exe
+* Implementing a prototype of this feature for dotnet.exe
+* Displaying the search results to users in a way that maximizes the visibility of the most relevant results
+
+## Solution
+
+The search feature leverages the search service provided by the NuGet API. Search targets multiple sources and displays all of their results one after the other. The command line interface is simple: __./NuGet.exe search \<query\>__.
+
+```
+PS C:\> ./NuGet.exe search logging
+```
+
+Each search result displays the name, version, number of downloads, and a preview of the description of the package.
+
+```
+====================
+Source: NuGet.org
+--------------------
+> Microsoft.Extensions.Logging.Abstractions | 5.0.0-preview.6.20305.6 | Downloads: 345,145,935
+Logging abstractions for Microsoft.Extensions.Logging.
+
+Commonly Used Types:
+Microsoft.Extensions.Lo...
+
+--------------------
+> Microsoft.Extensions.Logging | 5.0.0-preview.6.20305.6 | Downloads: 243,186,566
+Logging infrastructure default implementation for Microsoft.Extensions.Logging.
+When using NuGet 3....
+
+--------------------
+> Microsoft.IdentityModel.Logging | 6.6.0 | Downloads: 120,177,247
+Includes Event Source based logging support.
+
+--------------------
+> Microsoft.Extensions.Logging.Configuration | 5.0.0-preview.6.20305.6 | Downloads: 63,205,816
+```
+
+If a particular source returns no results for a query, it will display output that indicates this.
+
+```
+====================
+Source: dotnet-msbuild
+--------------------
+No results found.
+```
+
+### Optional Arguments
+
+Search provides users with some optional arguments that they can use to customize their search queries:
+
+| Name | Description | Usage |
+| ---  |     ---     |  :-:  |
+| PreRelease | `true` or `false` determining whether to include pre-release packages | -PreRelease `<true/false>` |
```

Comment:
nit: other commands like `install` and `list` have prerelease off by default and included with just a switch like `-Prerelease`. AFAIK you don't pass in `true` or `false` but I may be wrong here. (I'm just looking at `nuget.exe list -help` and `nuget.exe install -help`)

Reply:
You may find some option flags requiring true/false in nuget.exe, which is a bug (so don't copy everything you see there 😄). 
Maybe even rename this to `IncludePreRelease` ? Implying that the default is false.

Reply:
I've changed -PreRelease to just be a switch. I haven't renamed it to -IncludePreRelease simply to maintain consistency with other nuget commands like _list_.